src/Microsoft.ML.Data/Transforms/ConvertTransform.cs

```diff
@@ -169,6 +179,25 @@ private static VersionInfo GetVersionInfo()
         // This is parallel to Infos.
         private readonly ColInfoEx[] _exes;
 
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="name">Name of the output column.</param>
+        /// <param name="source">Name of the column to be converted.  If this is null '<paramref name="name"/>' will be used.</param>
+        /// <param name="resultType">The expected type of the converted column.</param>
+        /// <param name="keyRange">For a key column, this defines the range of values.</param>
+        public ConvertTransform(IHostEnvironment env,
+            IDataView input,
+            string name,
+            string source = null,
+            DataKind? resultType = null,
```

Comment:
>DataKind? [](start = 12, length = 9)

Hi @zeahmed thanks for doing this. Quick question, why is this nullable? What happens if it is null?

Reply:
Actually this is interesting. Unlike some other transforms, this transform is one of those where we kind of do have a required argument. Hmmm. We need to think carefully about how we want to phrase the constructor here. We may need to refine my prior idea from #380 of having the signature be `string name, string source = null`. That may have been an error.

Reply:
Or maybe not. Maybe just putting them before `name` is fine, maybe. But that's kind of weird too, since then you have the input data view, *then* required parameters, then names, then parameters with defaults. But maybe.

Reply:
Regarding `resultType` being null. I see following happening in the code. The decision is being made based on resultType, KeyRange and Range parameters. At the end, if everything is null `DataKind.Num` is the value.

https://github.com/zeahmed/machinelearning/blob/16f7883933b56f8fd86077bf0fd262b24374e9d0/src/Microsoft.ML.Data/Transforms/ConvertTransform.cs#L211